src/syscheckd/fim_db.c

```diff
@@ -531,27 +532,34 @@ int fim_db_exec_simple_wquery(fdb_t *fim_sql, const char *query) {
 }
 
 int fim_db_sync_path_range(fdb_t * fim_sql, pthread_mutex_t *mutex, fim_tmp_file *file, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_callback_sync_path_range,
-            storage, (void *) (int) 0);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_callback_sync_path_range, storage,
+                                    NULL, NULL, NULL);
 }
 
 int fim_db_delete_not_scanned(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path,
-            storage, (void *) (int) 1);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path, storage,
+                                    (void *) true, (void *) FIM_SCHEDULED, NULL);
 }
 
 int fim_db_delete_range(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path,
-            storage, (void *) (int) 0);
+    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_remove_path, storage,
+                                    (void *) false, (void *) FIM_SCHEDULED, NULL);
 }
 
-int fim_db_process_missing_entry(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage, fim_event_mode mode) {
-    return fim_db_process_read_file(fim_sql, file, mutex, fim_db_process_path, storage, (void *) (fim_event_mode) mode);
+int fim_db_process_missing_entry(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage, fim_event_mode mode, whodata_evt * w_evt) {
```

Comment:
Isn't it possible to merge the functions `fim_db_delete_not_scanned`, `fim_db_delete_range` and `fim_db_process_missing_entry` into a single one by adding an `alert` parameter?

Reply:
Thanks @TomasTurina , 

We have decided to name the wrapper functions differently because they perform different operations, so we can easily identify the action performed by each one of them.